multiversx_sdk/network_providers/api_network_provider.py

```diff
@@ -181,7 +179,7 @@ def get_token_of_account(self, address: Address, token: Token) -> TokenAmountOnN
         Able to handle both fungible and non-fungible tokens (NFTs, SFTs, MetaESDTs).
         """
         if token.nonce:
-            nonce_as_hex = decimal_to_padded_hex(token.nonce)
+            nonce_as_hex = TokenComputer().convert_nonce_to_hex(token)
```

Comment:
Instead of this and the concat / format below, can we call `TokenComputer.compute_extended_identifier(token)`?